bootable-jar/WFLY-14411_Bootable_Jar_Upgrade.adoc

```diff
@@ -0,0 +1,215 @@
+feature-pack artifact has been added to the dependencies. In this case the version is being retrieved from the dependencies. If the version
+is missing and no Feature-pack artifact is retrieved in the dependencies, a failure occurs during packaging.
+
+* Third party galleon feature-packs benefit from this upgrade capability for JBoss Module artifacts they are bringing to the provisioned server.
+
+* The server jar components that can be upgraded are:
+** The JBoss module runtime jar (jboss-modules.jar file).
+** All jar artifacts referenced from JBoss Modules modules.
+
+* The set of artifacts that can be upgraded can be retrieved by setting the parameter `<dump-original-artifacts>true</dump-original-artifacts> or 
+the system property `bootable.jar.dump.original.artifacts` to true when building a bootable JAR. 
+The file `target/bootable-jar-build-artifacts/bootable-jar-server-original-artifacts.xml` is generated. 
+It contains XML elements for the Galleon feature-packs dependencies, JBoss Modules runtime and artifacts. 
+JBoss Modules modules artifacts are grouped by JBoss Modules name.
+The generated file contains only the artifacts that are provisioned by Galleon. 
+Each artifact version is the one that would get installed when building the Bootable JAR without upgrade.
+
+* An artifact upgraded to the same version as the one referenced in the Galleon feature-pack is not upgraded. In this case a warning is displayed during build.
+
+* It is possible to downgrade an artifact to an older version. In this case a warning is displayed during build. 
+Warning can be disabled with `<disable-warn-for-artifact-downgrade>true</disable-warn-for-artifact-downgrade>` element.
+
+Example of an hypotetical `undertow-core` and `wildfly-ee-galleon-pack` upgrade:
+
+```
+...
+        <dependency>
+            <groupId>io.undertow</groupId>
+            <artifactId>undertow-core</artifactId>
+            <version>2.2.4</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly</groupId>
+            <artifactId>wildfly-ee-galleon-pack</artifactId>
+            <version>23.0.1.Final</version>
+            <type>zip</type>
+            <scope>provided</scope>
+        </dependency>
+...
+
+<plugin>
+        <groupId>org.wildfly.plugins</groupId>
+        <artifactId>wildfly-jar-maven-plugin</artifactId>
+        <configuration>
+            <feature-packs>
+                <feature-pack>
+                    <groupId>org.wildfly</groupId>
+                    <artifactId>wildfly-galleon-pack</artifactId>
+                    <version>23.0.0.Final</version>
+                </feature-pack>
+            </feature-packs>
+            <layers>
+                <layer>jaxrs-server</layer>
+            </layers>
+            <!-- We list the set of artifacts we want to see replaced during provisioning -->
+            <overriden-server-artifacts>
```

Comment:
If `overriden-server-artifacts` is the final name, should be this updated across the document?

Reply:
Thank-you, I fixed the typos.